### PR TITLE
Suppress telemetry feature

### PR DESCRIPTION
**Pull request by eliasyishak**
Addresses issue:
- https://github.com/dart-lang/tools/issues/124

This feature allows us to call a method on the analytics instance and suppress any events from being sent for the current invocation. This will allow clients of this tool such as the flutter-tool to simply call the method for a given command such as `flutter --suppress-analytics`

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: pkgs/unified_analytics/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ## 3.0.1-wip
 
 - Enhanced `LogFileStats` data to include information about flutter channel counts and tool counts
+- Added new method to suppress telemetry collection temporarily for current invocation via `analytics.suppressTelemetry()`
```
Comment (christopherfujino) [inline pkgs/unified_analytics/CHANGELOG.md:4]: I think this should bump the minor version, since we're adding a new method.

Comment (eliasyishak) [reply pkgs/unified_analytics/CHANGELOG.md:4]: Wow you read my mind! :D i just pushed that update once you commented

Comment (christopherfujino) [reply pkgs/unified_analytics/CHANGELOG.md:4]: Well actually, is anyone implementing the `Analytics` class? If so, this is a breaking change, right?

File: pkgs/unified_analytics/lib/src/analytics.dart [hunk 2]
```diff
@@ -224,6 +224,13 @@ abstract class Analytics {
   /// Setting the telemetry status will also send an event to GA
   /// indicating the latest status of the telemetry from [reportingBool]
   Future<void> setTelemetry(bool reportingBool);
+
+  /// Calling this will result in telemetry collection being suppressed for
+  /// the current invocation
```
Comment (christopherfujino) [inline pkgs/unified_analytics/lib/src/analytics.dart:229]:
```suggestion
  /// the current invocation.
```

Comment (christopherfujino) [reply pkgs/unified_analytics/lib/src/analytics.dart:229]: it's a nit you don't have to fix here, but really all dartdocs should be complete sentences, ending in periods.

Comment (eliasyishak) [reply pkgs/unified_analytics/lib/src/analytics.dart:229]: Ah good point, going to create a separate issue for this
- https://github.com/dart-lang/tools/issues/126

File: pkgs/unified_analytics/test/suppression_test.dart [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  const flutterChannel = 'flutterChannel';
+  const flutterVersion = 'flutterVersion';
+  const dartVersion = 'dartVersion';
+  const platform = DevicePlatform.macos;
+
+  final testEvent = Event.hotReloadTime(timeMs: 50);
+
+  setUp(() {
+    // Setup the filesystem with the home directory
+    final fsStyle =
```
Comment (christopherfujino) [inline pkgs/unified_analytics/test/suppression_test.dart:34]: what style guide are you following for this codebase?

Comment (eliasyishak) [reply pkgs/unified_analytics/test/suppression_test.dart:34]: I was originally following what we did on the flutter side but this PR a few weeks ago had a big diff
- https://github.com/dart-lang/tools/pull/108

Comment (christopherfujino) [reply pkgs/unified_analytics/test/suppression_test.dart:34]: So what's the policy? Because I'm seeing both styles in the repo.

**Review by christopherfujino — APPROVED** [review]
Code LGTM, just wanna make sure if this is breaking change we bump the major version.

File: pkgs/unified_analytics/test/suppression_test.dart (hunk 3 above)
Comment (eliasyishak) [reply pkgs/unified_analytics/test/suppression_test.dart:34]: I'm still following how we do things on the flutter side while fixing for any of the lints that come up, like the removing the type from the left side when assigning variables.

If you see anywhere else that needs fixing, i'm happy to fix

**Pull request merged**